dev/MAINTAINING.md

```diff
@@ -26,71 +26,98 @@
-    * Test latest GHC version with -O0 and -O1
+    * Test prime GHC version with -O0 and -O1
 
 * _Benchmarks_:
 
     * Check regressions from previous release
-    * Run benchmarks with large stream size (`bench.sh -- long`) to
+    * Run benchmarks with large stream size (`bench.sh --long`) to
       check for space leaks and to ensure constant memory usage for streaming
       operations.
     * Run benchmarks with `dev` flag on. Some `fileio` benchmarks are disabled
       in regular runs.
     * Check comparative benchmarks using streaming-benchmarks
 
-* _Examples_:
+* _Dependencies_:
+
+    * Make sure all dependency bounds can use latest versions
+    * Update `stack.yaml` to latest stable resolver, cleanup extra-deps
+    * Make sure fusion-plugin is up to date and uploaded
+    * Check the dependency footprint of the library. See
+      [Generating Dependency Graphs](##generating-dependency-graphs) section.
+
+* _Dependent Packages_:
 
     * Update
       [streamly-examples](https://github.com/composewell/streamly-examples)
       to make sure it runs with the latest release.
     * Check the performance of examples where applicable
+    * Update streamly-bytestring
+    * Consider updating other packages e.g. streamly-process, streamly-lz4
 
 * _Update Package Metadata:_
 
-    * Update `stack.yaml` to latest stable resolver, cleanup extra-deps
     * Make sure the description in cabal file is in sync with README and other docs
     * Make sure CI configs include last three major releases of GHC in CI testing.
-    * Update `tested-with` field
-    * Make sure all dependency bounds can use latest versions
+    * Update GHC `tested-with` field
+    * Update `docs/building.md` with the distributions tested with
     * Make sure any additional files are added to `extra-source-files` in cabal
-      file
+      file. Artifacts required for build, test, benchmarks, docs, licenses
+      should be packaged. Build environment customization may or may not be
+      packaged.
 
-* Copyrights and Contibutors
+* _Copyrights and Contibutors_
 
     * Make sure contributors to the release are listed in
       `credits/CONTRIBUTORS.md`.
     * Bump the release version in `credits/CONTRIBUTORS.md`.
     * Make sure any third party code included in the release has been listed in
       `credits/COPYRIGHTS.md` and the license is added to the repo.
-    * Change the `Unreleased` section, if exists, at the top of
-      `credits/COPYRIGHTS.md` to the new release version number.
+    * Bump the release version in `credits/COPYRIGHTS.md`.
 
 * _Update changelog & Version_:
 
+    * Find API changes using `cabal-diff streamly 0.8.0 .` and record
+      them in docs/API-changelog.txt.
     * Make sure all the bug fixes being included in this release are marked
       with a target release on github. So that users can search by release if
       they want.
+    * Bump the package version in configure.ac and run autoreconf
```

Comment:
We should try to automate the bumping package version. We can automate a lot of things `cabal-diff` etc. But we can start with this and slowly improve the release process.

Reply:
All this can be done via  a script, but making that handle all cases/errors would require some effort.